pkg/ingester/recovery.go

```diff
@@ -220,9 +220,17 @@ func (r *ingesterRecoverer) Close() {
 
 	for _, inst := range r.ing.getInstances() {
 		inst.forAllStreams(context.Background(), func(s *stream) error {
+			s.chunkMtx.Lock()
+			defer s.chunkMtx.Unlock()
```

Comment:
I traced through some of the paths that are taken by ConvertHead below but it branches out a little bit. I'm wondering if it would be safer to explicitly just lock and unlock around the conditional below rather than using defer?

Reply:
Given that the new code also does a return, I guess weigh this against having to unlock in 2 places vs defer. six and one half dozen....